### PR TITLE
add startEditWithCurrentModalTool

### PR DESCRIPTION
**Pull request by pineapplespatula**
one more time

### CODE REVIEW

File: src/ps/ui.js [hunk 1]
```diff
@@ -399,8 +399,8 @@ define(function (require, exports, module) {
      *
      * @return {Promise}
      */
-    UI.prototype.startEditWithCurrentModalTool = function () {
```
Comment (volfied) [inline src/ps/ui.js]: JSDOC please

File: src/ps/ui.js [hunk 2]
```diff
@@ -396,11 +396,12 @@ define(function (require, exports, module) {
...
     /**
      * Starts a modal edit state with the current active tool
-     *
+     * 
+     * @param {function} callback this will be called in the case of an error
      * @return {Promise}
      */
-    UI.prototype.startEditWithCurrentModalTool = function () {
-        return _ui.startEditWithCurrentModalToolAsync();
+    UI.prototype.startEditWithCurrentModalTool = function (callback) {
```
Comment (iwehrman) [inline src/ps/ui.js]: I'm confused.

Comment (iwehrman) [inline src/ps/ui.js]: What is this callback for? The `Async` version replaces the callback parameter and gives you a function that returns a promise instead. Unless there is something very unusual about this native function, I don't think a callback is needed here.

Comment (iwehrman) [inline src/ps/ui.js]: See here: https://github.com/petkaantonov/bluebird/blob/master/API.md#promisepromisifyallobject-target--object-options---object

**Comment by pineapplespatula** [comment]
@iwehrman  sounds good to me

**Comment by iwehrman** [comment]
As far as I know, yes.

**Pull request closed without merging**